gprofiler/profilers/java.py

```diff
@@ -321,7 +321,8 @@ def __init__(
         collect_meminfo: bool = True,
     ):
         self.process = process
-        self._stop_event = stop_event
+        self._profiler_state = profiler_state
+        self._stop_event = self._profiler_state.stop_event
```

Comment:
Don't store it on the class - access directly from the `_profiler_state.stop_event`.